RepositoryCompilerTests/Unit/CaDETCodeModelTests.cs

```diff
@@ -212,16 +212,80 @@ public void Calculates_method_variables()
+            var doctorService = classes.Find(c => c.Name.Equals("DoctorService"));
+            var otherDoctor = otherClasses.Find(c => c.Name.Equals("Doctor"));
+            var otherDateRange = otherClasses.Find(c => c.Name.Equals("DateRange"));
+
+            dateRange.FindField("testDictionary").GetLinkedTypes().ShouldContain(doctor);
+            dateRange.FindField("testDictionary").GetLinkedTypes().ShouldNotContain(otherDoctor);
+            dateRange.FindField("testDictionary").GetLinkedTypes().ShouldContain(doctorService);
+            doctor.FindField("TestDR").GetLinkedTypes().ShouldContain(dateRange);
+            doctor.FindField("TestDR").GetLinkedTypes().ShouldNotContain(otherDateRange);
+            doctorService.FindField("_doctors").GetLinkedTypes().ShouldContain(doctor);
+        }
+
+        [Fact]
+        public void Checks_method_linked_return_types()
+        {
+            CodeModelFactory factory = new CodeModelFactory(LanguageEnum.CSharp);
+
+            List<CaDETClass> classes = factory.CreateClassModel(_testDataFactory.GetMultipleClassTexts());
+            List<CaDETClass> otherClasses = factory.CreateClassModel(_testDataFactory.GetClassesFromDifferentNamespace());
+
+            var dateRange = classes.Find(c => c.Name.Equals("DateRange"));
+            var doctor = classes.Find(c => c.Name.Equals("Doctor"));
+            var doctorService = classes.Find(c => c.Name.Equals("DoctorService"));
+            var otherDoctor = otherClasses.Find(c => c.Name.Equals("Doctor"));
+            var otherDateRange = otherClasses.Find(c => c.Name.Equals("DateRange"));
+
+            doctor.FindMember("TestFunction").GetLinkedReturnTypes().ShouldContain(dateRange);
+            doctor.FindMember("TestFunction").GetLinkedReturnTypes().ShouldNotContain(otherDateRange);
+            doctorService.FindMember("FindAvailableDoctor").GetLinkedReturnTypes().ShouldContain(doctor);
+            doctorService.FindMember("FindAvailableDoctor").GetLinkedReturnTypes().ShouldNotContain(otherDoctor);
+        }
+
+        [Fact]
+        public void Checks_variable_linked_types()
+        {
+            CodeModelFactory factory = new CodeModelFactory(LanguageEnum.CSharp);
+
+            List<CaDETClass> classes = factory.CreateClassModel(_testDataFactory.GetMultipleClassTexts());
+            List<CaDETClass> otherClasses = factory.CreateClassModel(_testDataFactory.GetClassesFromDifferentNamespace());
+
+            var dateRange = classes.Find(c => c.Name.Equals("DateRange"));
+            var doctor = classes.Find(c => c.Name.Equals("Doctor"));
+            var doctorService = classes.Find(c => c.Name.Equals("DoctorService"));
+            var otherDoctor = otherClasses.Find(c => c.Name.Equals("Doctor"));
+            var otherDateRange = otherClasses.Find(c => c.Name.Equals("DateRange"));
+
+            doctor.FindMember("TestFunction").Variables.SelectMany(v => v.GetLinkedTypes()).ShouldContain(doctor);
```

Comment:
doctor.FindMember("TestFunction").Variables.SelectMany(v => v.GetLinkedTypes()) is duplicated 5 times. Change to:

var variables = doctor.FindMember("TestFunction").Variables.SelectMany(v => v.GetLinkedTypes());
variables.ShouldContain...